src/unlimited.cpp

```diff
@@ -1153,7 +1154,23 @@ bool IsChainNearlySyncd()
     LOCK(cs_ischainnearlysyncd);
     return fIsChainNearlySyncd;
 }
-
+uint64_t LargestBlockSeen(uint64_t nBlockSize)
+{
+     // return the largest block size that we have seen since startup
+     static boost::atomic<uint64_t> nLargestBlockSeen;
+     static bool fInit = true;
+     if (fInit) {
+         nLargestBlockSeen.store(BLOCKSTREAM_CORE_MAX_BLOCK_SIZE);
+         fInit = false;
+     }
+
+     uint64_t nSize = nLargestBlockSeen.load();
+     if (nBlockSize > nSize) {
+         nLargestBlockSeen.store(nBlockSize);
```

Comment:
this works just as it should...we're just swapping the old with the new if the blocksize has increased...not need for a loop here.